pkg/scheduler/framework/runtime/framework.go

```diff
@@ -686,7 +686,9 @@ func (f *frameworkImpl) RunPreFilterPlugins(ctx context.Context, state *framewor
 			if len(pluginsWithNodes) == 1 {
 				msg = fmt.Sprintf("node(s) didn't satisfy plugin %v", pluginsWithNodes[0])
 			}
-			return nil, framework.NewStatus(framework.Unschedulable, msg)
+			status := framework.NewStatus(framework.Unschedulable, msg)
+			status.SetPlugin(pluginsWithNodes...)
```

Comment:
We should record pluginsWithNodes no matter we have a empty or non-empty `result.NodeNames`.

Reply:
as @sanposhiho [said](https://github.com/kubernetes/kubernetes/issues/118369#issuecomment-1581129204), we just need record `PreFilter` plugins when `PreFilterResult` have no `NodeNames`

Reply:
Not `no NodeNames`, we need to record a plugin if it filters out some nodes(i.e. not AllNodes). And we have to do this per plugin, don't need to consider the result combined.

Let's say we have this case:
Two plugins, two nodes.
PluginA will filter out nodeA in prefilter and filter.
PluginB will pass both nodes.

1. After prefilter, we will have a non-nil result having nodeA in nodeName.
2. In currently implementation, we'll have no plugin registered as unschedulablePlugins.
3. We continue to filter, although PluginB pass nodeA, PluginA will reject it. Thus make this pod unscheduable.
4. In this case, when pod became unschedulable. An event related to pluginA could make this pod more scheduable. Thus we need to record pluginA for queueHint.

Reply:
please review the lastest commit,  confirm whether it is the correct way to implement you saied

Reply:
Actually, I think just use `status.SetPlugin(pluginsWithNodes...)` in defer in L662 would be sufficient

Reply:
maybe not sufficient.  there is no need register  unscheduable  prefilter plugin for deferent node(`Diagnosis.NodeToStatusMap` field) like [in filter phase](https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/schedule_one.go#L593) ?

Reply:
> Actually, I think just use `status.SetPlugin(pluginsWithNodes...)` in defer in L662 would be sufficient

can't cover the [case](https://github.com/kubernetes/kubernetes/pull/122251#pullrequestreview-1798137223)